contrib/nvim_api_docs.py

```diff
@@ -0,0 +1,527 @@
+#!/usr/bin/env python
+"""Parses Doxygen XML output to generate Neovim's API documentation.
```

Comment:
`contrib/` is more for unsupported nice-to-have stuff. Probably this should live in:

    scripts/gen_api_vimdoc.py